VisualizationBase/src/nodes/InfoNode.h

```diff
@@ -0,0 +1,116 @@
+**
+** Copyright (c) 2015 ETH Zurich
+** All rights reserved.
+**
+** Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+** following conditions are met:
+**
+**    * Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+**      disclaimer.
+**    * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+**      following disclaimer in the documentation and/or other materials provided with the distribution.
+**    * Neither the name of the ETH Zurich nor the names of its contributors may be used to endorse or promote products
+**      derived from this software without specific prior written permission.
+**
+**
+** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+** INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+** WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+**
+***********************************************************************************************************************/
+#pragma once
+
+#include "../VisualizationBase/src/visualizationbase_api.h"
+#include "ModelBase/src/nodes/Node.h"
+#include "ModelBase/src/nodes/TypedList.h"
+#include "ModelBase/src/nodes/nodeMacros.h"
+
+DECLARE_TYPED_LIST(VISUALIZATIONBASE_API, Visualization, InfoNode)
+
+namespace Visualization {
+
+/**
+ * An InfoNode is the base class of all nodes which display some sort of statistics
+ * on a target node. There is a register which keeps track of all possible informations
+ * to be displayed.
+ */
+class VISUALIZATIONBASE_API InfoNode : public Super<Model::Node>
+{
+	DECLARE_TYPE_ID
+
+	public:
+		typedef QString (*InfoGetter) (Model::Node* target);
+
+		InfoNode(Model::Node* target);
+		InfoNode(Model::Node *target, QList<QString> enabledInfos);
+		virtual ~InfoNode();
+
+		const Model::Node* target() const;
+
+		/**
+		 * If the statistics can be represented as a string, use
+		 * this to represent them as HTML text. Then the same visualization
+		 * can be used for multiple statistics nodes.
+		 */
+		QString infoHtml() const;
+		/**
+		 * Updates the node's stored information.
+		 * @param isAutoUpdate Is this an automatic update, or a forced update?
+		 *					   A forced update will update everything.
+		 */
+		virtual void updateInfo(bool isAutoUpdate);
```

Comment:
Coming back to what I wrote above, a forced update should happen only when the user request it, and so should be called directly from the handler. In all other cases you should use the usual update procedure.